lib/exe/css.php

```diff
@@ -344,6 +360,37 @@ function css_fixreplacementurls($replacements, $location) {
 }
 
 /**
+ * Wrapper for the files, content and mediatype for the event CSS_STYLES_INCLUDED
+ *
+ * @param string $mediatype type ofthe current media files/content set
+ * @param array $files set of files that define the current mediatype
+ * @return array
+ */
+function css_filewrapper($mediatype, $files=array()){
+    return array(
+            'files'                 => $files,
+	        'mediatype'             => $mediatype,
+	        'encapsulate'           => in_array($mediatype, array('screen', 'print')),
```

Comment:
I got another idea... What about completely new `style.ini` files per context? I think that makes the most sense as you essentially want to rebuild the style depending on different parameters.